encode.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+
+	vegeta "github.com/tsenart/vegeta/lib"
+)
+
+const (
+	encodingCSV  = "csv"
+	encodingGob  = "gob"
+	encodingJSON = "json"
+)
+
+type decoderFunc func(io.Reader) vegeta.Decoder
+type encoderFunc func(io.Writer) vegeta.Encoder
+
+func encodeCmd() command {
+	var (
+		fs     = flag.NewFlagSet("encode", flag.ExitOnError)
+		from   = fs.String("from", "gob", "Input decoding [csv, gob, json]")
+		to     = fs.String("to", "json", "Output encoding [csv, gob, json]")
+		output = fs.String("output", "", "Output file")
+	)
+
+	fs.Usage = func() {
+		fmt.Println("Usage: vegeta encode [flags] [<file>...]")
+		fs.PrintDefaults()
+	}
+
+	return command{
+		fs,
+		func(args []string) error {
+			fs.Parse(args)
+			return encode(*from, *to, *output, args...)
+		},
+	}
+}
+
+func encode(from, to, output string, inputs ...string) error {
+	var decFn decoderFunc
+
+	switch from {
+	case encodingCSV:
+		decFn = vegeta.NewCSVDecoder
+	case encodingJSON:
+		decFn = vegeta.NewJSONDecoder
+	default:
+		// Gob is our default decode format to play nicely in pipes with attack.
+		decFn = vegeta.NewDecoder
+	}
+
+	decs := []vegeta.Decoder{}
+
+	if len(inputs) > 0 {
+		for _, name := range inputs {
+			f, err := os.Open(name)
+			if err != nil {
+				return err
+			}
+			defer f.Close()
+
+			decs = append(decs, decFn(f))
+		}
+	} else {
+		decs = append(decs, decFn(os.Stdin))
+	}
+
+	in := vegeta.NewRoundRobinDecoder(decs...)
```

Comment:
I am wondering what makes most sense to do. Shall we read results in a round-robin fashion and re-encode each out as they come or should we instead read all of them, sort them and then write them out. Perhaps this should be a flag.

Reply:
What is the upside of the sorting? And sorting per input, over all inputs, by what criteria?

Reply:
I think we can delegate this to external tools, indeed.

```
cat results.bin | vegeta encode | jq -s 'sort_by(.timestamp)'
```